scripts/charmm_to_fah.py

```diff
@@ -0,0 +1,309 @@
+del(platform)
+del(simulation.context)
+del(simulation)
+simulation = None
+
+if verbose: print("Creating constrained OpenMM system...")
+coords = modeller.positions
+min_crds = [coords[0][0], coords[0][1], coords[0][2]]
+max_crds = [coords[0][0], coords[0][1], coords[0][2]]
+
+for coord in coords:
+    min_crds[0] = min(min_crds[0], coord[0])
+    min_crds[1] = min(min_crds[1], coord[1])
+    min_crds[2] = min(min_crds[2], coord[2])
+    max_crds[0] = max(max_crds[0], coord[0])
+    max_crds[1] = max(max_crds[1], coord[1])
+    max_crds[2] = max(max_crds[2], coord[2])
+
+psf.setBox(max_crds[0]-min_crds[0],
+           max_crds[1]-min_crds[1],
+           max_crds[2]-min_crds[2],
+)
+system = psf.createSystem(params, nonbondedMethod=nonbonded_method, nonbondedCutoff=nonbonded_cutoff, constraints=app.HBonds)
+if verbose: print("Adding barostat...")
+system.addForce(openmm.MonteCarloBarostat(pressure, temperature, barostat_frequency))
+
+# Create simulation.
+if verbose: print("Creating simulation...")
+integrator = openmm.LangevinIntegrator(temperature, collision_rate, timestep)
+#platform = openmm.Platform.getPlatformByName('CPU')
+platform = openmm.Platform.getPlatformByName('OpenCL')
+platform.setPropertyDefaultValue('OpenCLPrecision', 'double') # use double precision
+simulation = app.Simulation(modeller.topology, system, integrator, platform=platform)
+simulation.context.setPositions(modeller.positions)
+
+# Write modeller positions.
+if verbose: print("Writing modeller output...")
+filename = os.path.join(workdir, 'modeller.pdb')
+positions = simulation.context.getState(getPositions=True).getPositions(asNumpy=True)
+print(abs(positions / unit.nanometers).max())
+app.PDBFile.writeFile(simulation.topology, positions, open(filename, 'w'))
+
+# Minimize energy.
+if verbose: print("Minimizing energy...")
+potential_energy = simulation.context.getState(getEnergy=True).getPotentialEnergy()
+if numpy.isnan(potential_energy / unit.kilocalories_per_mole):
+    raise Exception("Potential energy is NaN before minimization.")
+if verbose: print("Initial potential energy : %10.3f kcal/mol" % (potential_energy / unit.kilocalories_per_mole))
+simulation.minimizeEnergy(maxIterations=max_minimization_iterations)
+potential_energy = simulation.context.getState(getEnergy=True).getPotentialEnergy()
+if numpy.isnan(potential_energy / unit.kilocalories_per_mole):
+    raise Exception("Potential energy is NaN after minimization.")
+if verbose: print("Final potential energy:  : %10.3f kcal/mol" % (potential_energy / unit.kilocalories_per_mole))
+
+
+# Write initial positions.
+filename = os.path.join(workdir, 'minimized.pdb')
+positions = simulation.context.getState(getPositions=True).getPositions()
+app.PDBFile.writeFile(simulation.topology, positions, open(filename, 'w'))
+
+# Assign temperature
+simulation.context.setVelocitiesToTemperature(temperature)
+
+# Take a few steps to relax structure.
+if verbose: print("Taking a few steps...")
+simulation.step(nsteps)
```

Comment:
We can actually minimize again after we take a few steps with a very tiny timestep if things are still blowing up.